spec/requests/api/v3/courses/index_spec.rb

```diff
@@ -257,6 +257,45 @@
     end
   end
 
+  describe "provider filter" do
+
+    context "with a provider specified" do
+      let(:request_path) { "/api/v3/courses?filter[provider.provider_name]=#{filtered_provider_course.provider.provider_name}" }
+      let(:provider_filtered_by) { create(:provider) }
+      let(:filtered_provider_course) { create(:course, provider: provider_filtered_by, site_statuses: [findable_status], enrichments: [published_enrichment]) }
+
+      before do
+        provider_filtered_by
+        filtered_provider_course
+      end
+
+      it "is returned" do
+        get request_path
+        json_response = JSON.parse(response.body)
+        course_hashes = json_response["data"]
+        expect(course_hashes.count).to eq(1)
+      end
+    end
+
+    context "with a different provider specified" do
+      let(:request_path) { "/api/v3/courses?filter[provider.provider_name]=a+different+provider" }
+      let(:provider_excluded) { create(:provider) }
+      let(:excluded_provider_course) { create(:course, provider: provider_excluded, site_statuses: [build(:site_status, :findable)], enrichments: [published_enrichment]) }
```

Comment:
Did you end up seeing if you could get rid of these attributes?

Reply:
I did manage to get rid of some, but the two left here are required for a course to show up in a search. The scoping will remove them otherwise